File: array/print_distinct.cpp
#include <bits/stdc++.h>

using namespace std;


int main() {
    int arr[] = {1, 2, 3, 3, 4, 4, 5}, n = 7;

    unordered_set<int> set;

    for (int i = 0; i < 7; i ++) {
        if (!set.count(arr[i])) {
            set.insert(arr[i]);
            cout << arr[i] << " ";
        }
    }

    return 0;
}

File: array/print_distinct.js
var arr = [1, 2, 3, 3, 4, 4, 5];

var new_arr = [];
var set = new Set();
for (let i in arr) {
    if (!set.has(arr[i])) {
        set.add(arr[i]);
        new_arr.push(arr[i]);
    } 
}

console.log(new_arr);